cmd/synapse/bin.go

```diff
@@ -138,6 +141,21 @@ func run(cmd *cobra.Command, args []string) {
 			case <-time.After(global.GracefulTimeout):
 				logger.Errorf("Graceful timeout exceeded. Brutally killing the application")
 			}
+		}
+
+	case <-connectionFailed:
```

Comment:
how are re-trying case of connection failed ?

Reply:
This will be executed if max attempt to connection exhausted & failed